code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -1,281 +1,271 @@
-	attached++
-	//ensure we detach once we no longer need to be attached
-	addtimer(CALLBACK(src, .proc/detach), MAX_IMPREGNATION_TIME)
-
-
-	if(!sterile)
-		M.take_bodypart_damage(strength,0) //done here so that humans in helmets take damage
-		M.Unconscious(MAX_IMPREGNATION_TIME/0.3) //something like 25 ticks = 20 seconds with the default settings
-
-	GoIdle() //so it doesn't jump the people that tear it off
-
-	addtimer(CALLBACK(src, .proc/Impregnate, M), rand(MIN_IMPREGNATION_TIME, MAX_IMPREGNATION_TIME))
-
-/obj/item/clothing/mask/facehugger/proc/detach()
-	attached = 0
-
-/obj/item/clothing/mask/facehugger/proc/Impregnate(mob/living/target)
-	if(!target || target.stat == DEAD) //was taken off or something
-		return
-
-	if(iscarbon(target))
-		var/mob/living/carbon/C = target
-		if(C.wear_mask != src)
-			return
-
-	if(!sterile)
-		target.visible_message(span_danger("[src] falls limp after violating [target]'s face!"), \
-								span_userdanger("[src] falls limp after violating your face!"))
-
-		Die()
-		icon_state = "[base_icon_state]_impregnated"
-		worn_icon_state = "[base_icon_state]_impregnated"
-
-		var/obj/item/bodypart/chest/LC = target.get_bodypart(BODY_ZONE_CHEST)
-		if((!LC || LC.status != BODYPART_ROBOTIC) && !target.getorgan(/obj/item/organ/body_egg/alien_embryo))
-			new /obj/item/organ/body_egg/alien_embryo(target)
-			var/turf/T = get_turf(target)
-			log_game("[key_name(target)] was impregnated by a facehugger at [loc_name(T)]")
-
-	else
-		target.visible_message(span_danger("[src] violates [target]'s face!"), \
-								span_userdanger("[src] violates your face!"))
-
-/obj/item/clothing/mask/facehugger/proc/GoActive()
-	if(stat == DEAD || stat == CONSCIOUS)
-		return
-
-	stat = CONSCIOUS
-	icon_state = "[base_icon_state]"
-	worn_icon_state = "[base_icon_state]"
-
-/obj/item/clothing/mask/facehugger/proc/GoIdle()
-	if(stat == DEAD || stat == UNCONSCIOUS)
-		return
-
-	stat = UNCONSCIOUS
-	icon_state = "[base_icon_state]_inactive"
-	worn_icon_state = "[base_icon_state]_inactive"
-
-	addtimer(CALLBACK(src, .proc/GoActive), rand(MIN_ACTIVE_TIME, MAX_ACTIVE_TIME))
-
-/obj/item/clothing/mask/facehugger/proc/Die()
-	if(stat == DEAD)
-		return
-
-	icon_state = "[base_icon_state]_dead"
-	worn_icon_state = "[base_icon_state]_dead"
-	inhand_icon_state = "facehugger_inactive"
-	stat = DEAD
-
-	visible_message(span_danger("[src] curls up into a ball!"))
-
-/proc/CanHug(mob/living/M)
-	if(!istype(M))
-		return FALSE
-	if(M.stat == DEAD)
-		return FALSE
-	if(M.getorgan(/obj/item/organ/alien/hivenode))
-		return FALSE
-	var/mob/living/carbon/C = M
-	if(ishuman(C) && !(ITEM_SLOT_MASK in C.dna.species.no_equip))
-		var/mob/living/carbon/human/H = C
-		if(H.is_mouth_covered(head_only = 1))
-			return FALSE
-		return TRUE
-	return FALSE
-
-#undef MIN_ACTIVE_TIME
-#undef MAX_ACTIVE_TIME
-
-#undef MIN_IMPREGNATION_TIME
-#undef MAX_IMPREGNATION_TIME
+	var/mob/living/simple_animal/hostile/facehugger/mob_lamarr = new(src)
```

Comment:
it'd probably be best to make lamarr his own subtype

otherwise, an RD could use a regenerative black crossbreed from xenobio to make a non-sterile facehugger corpse from lamarr, then lazarus injector it to create a living, non-sterile facehugger.

... unless you want to promote that into being an intentional feature, of course.

Reply:
Jesus fucking Christ, whyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy

Reply:
![image](https://user-images.githubusercontent.com/47086570/127962100-502bb380-5e40-40cf-97a5-86f166d2b8d5.png)